outlines/prompts.py

```diff
@@ -19,12 +21,12 @@ class Prompt:
 
     """
 
-    template: str
-    signature: inspect.Signature
+    template: jinja2.Template
+    signature: Optional[inspect.Signature]
 
     def __post_init__(self):
-        self.parameters: List[str] = list(self.signature.parameters.keys())
-        self.jinja_environment = create_jinja_template(self.template)
+        if self.signature is not None:
+            self.parameters: List[str] = list(self.signature.parameters.keys())
```

Comment:
I didn't find where `self.parameters` is used in the logic, with the exception of tests, wonder if we can drop it? 🤔

Reply:
Yes, agreed! I'll check for usages and remove it :)